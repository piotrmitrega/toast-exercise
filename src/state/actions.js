
export const RECEIVE_INCOMING_ITEM = 'submissions/receive-incoming-item';

export const receiveIncomingItem = item => ({
  type: RECEIVE_INCOMING_ITEM,
  item
});
