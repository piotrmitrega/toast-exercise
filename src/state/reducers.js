import { combineReducers } from 'redux';

import { RECEIVE_INCOMING_ITEM } from './actions';

export const INITIAL_STATE = {
  incomingItems: []
};

export const submissions = (state = INITIAL_STATE, action) => {
  if (action.type === RECEIVE_INCOMING_ITEM) {
    const { item } = action;
    const { incomingItems } = state;

    return {
      ...state,
      incomingItems: [...incomingItems, item]
    };
  }

  return state;
};

export const reducers = combineReducers({
  submissions
});
