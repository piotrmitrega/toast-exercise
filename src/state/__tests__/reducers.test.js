import { RECEIVE_INCOMING_ITEM, receiveIncomingItem } from '../actions';
import { INITIAL_STATE, submissions } from '../reducers';

describe('submissions reducers tests suite', () => {
  describe(RECEIVE_INCOMING_ITEM, () => {
    test('append incoming item to the list', () => {
      const action = receiveIncomingItem({
        id: 1,
        data: { foo: 'bar' }
      });

      const state = {
        ...INITIAL_STATE,
        incomingItems: [{
          id: 0,
          data: {}
        }]
      };

      const result = submissions(state, action);

      expect(result).toMatchSnapshot();
    });
  });
});
