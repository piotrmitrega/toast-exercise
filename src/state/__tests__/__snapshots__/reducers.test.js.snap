// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`submissions reducers tests suite submissions/receive-incoming-item append incoming item to the list 1`] = `
Object {
  "incomingItems": Array [
    Object {
      "data": Object {},
      "id": 0,
    },
    Object {
      "data": Object {
        "foo": "bar",
      },
      "id": 1,
    },
  ],
}
`;
